src/pages/Settings/components/NameInput/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { createStyles, makeStyles } from "@material-ui/core";
+import Button from "@material-ui/core/Button";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import TextField from "@material-ui/core/TextField";
+import React, { memo, useCallback, useEffect, useMemo } from "react";
+import {
+  GardenUpdateParams,
+  updateGardenData,
+} from "../../../../helpers/api/gardens/updateGardenData";
+import { Garden } from "../../../../models/garden.model";
+import { useApi } from "../../../../utils/api/useApi";
+import styles from "./NameInput.module.css";
+
+type Props = {
+  currentGardenName: string;
+  initialGardenName: string;
+  onNameInputChange: (
+    e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
+  ) => void;
+  setShowNameInput: React.Dispatch<React.SetStateAction<boolean>>;
+  setGardenName: React.Dispatch<React.SetStateAction<string | undefined>>;
+  garden: Required<Garden> | undefined;
+};
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    buttonWrapper: {
+      position: "relative",
+      display: "inline-block",
+    },
+    buttonProgress: {
+      position: "absolute",
+      top: "50%",
+      left: "50%",
+      marginTop: -12,
+      marginLeft: -12,
+    },
+  })
+);
+
+export const NameInput: React.FC<Props> = memo(
+  ({
+    currentGardenName,
```

Comment:
Name input for edit page

Reply:
I'm not sure the best way of doing this but currently for the name and description input you can only see the label when you click on the update button, and while it should be obvious what each input is, it might be better to always show the labels? Not sure if that can be done with the way Buttons are implemented in Material-UI though.